src/main/java/org/genomicsdb/reader/GenomicsDBFeatureReader.java

```diff
@@ -136,61 +126,36 @@ public void close() throws IOException {
      * @param end   end position, inclusive (1-based)
      * @return iterator over {@link htsjdk.variant.variantcontext.VariantContext} objects
      */
-    public CloseableTribbleIterator<T> query(final String chr, final int start, final int end) throws IOException {
-        Optional<Coordinates.ContigInterval> contigInterval = Optional.of(
-                Coordinates.ContigInterval.newBuilder().setContig(chr).setBegin(start).setEnd(end).build());
-        List<String> chromosomeIntervalArraysPaths = resolveChromosomeIntervalArraysPaths(contigInterval);
-        return new GenomicsDBFeatureIterator(this.loaderJSONFile, chromosomeIntervalArraysPaths, this.featureCodecHeader,
-                this.codec, chr, OptionalInt.of(start), OptionalInt.of(end), Optional.of(this.intervalsPerArray));
+    public CloseableTribbleIterator<T> query(final String chr, final int start, final int end) {
+        if (this.exportConfiguration.hasArrayName()) {
+            return new GenomicsDBFeatureIterator(this.loaderJSONFile, 
+                    this.exportConfiguration, Optional.empty(), 
+                    this.featureCodecHeader, this.codec);
+        }
+        else {
+            Optional<Coordinates.ContigInterval> contigInterval = Optional.of(
+                    Coordinates.ContigInterval.newBuilder().setContig(chr).setBegin(start).setEnd(end).build());
+            List<String> chromosomeIntervalArraysPaths = 
+                    resolveChromosomeArrayFolderList(contigInterval);
+            return new GenomicsDBFeatureIterator(this.loaderJSONFile, 
+                    this.exportConfiguration, Optional.of(chromosomeIntervalArraysPaths),
+                    this.featureCodecHeader, this.codec, chr, OptionalInt.of(start), OptionalInt.of(end));
+        }
```

Comment:
Didn't really review this file - I'm assuming it works. This part of the code gives me nightmares.
Let me know if you need me to look at this.

Reply:
Hah - well genomicsdb tests pass anyway. I should try the gatk tests as well just to be safe...

Reply:
Yes please, try the gatk tests just to be sure we are maintaining compatibility.